homeassistant/components/squeezebox/__init__.py

```diff
@@ -1,31 +1,80 @@
 """The Squeezebox integration."""
 
+from asyncio import timeout
 import logging
 
+from pysqueezebox import Server
+
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import Platform
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    Platform,
+)
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
-from .const import DISCOVERY_TASK, DOMAIN, PLAYER_DISCOVERY_UNSUB
+from .const import (
+    CONF_HTTPS,
+    DATA_SERVER,
+    DISCOVERY_TASK,
+    DOMAIN,
+    PLAYER_DISCOVERY_UNSUB,
+    STATUS_API_TIMEOUT,
+    STATUS_QUERY_LIBRARYNAME,
+    STATUS_QUERY_UUID,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
 PLATFORMS = [Platform.MEDIA_PLAYER]
 
-
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up Squeezebox from a config entry."""
+    """Set up an LMS Server from a config entry."""
+    config = entry.data
+    session = async_get_clientsession(hass)
+    _LOGGER.debug("Reached async_setup_entry for host=%s(%s)", config[CONF_HOST],entry.entry_id)
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    https = config.get(CONF_HTTPS, False)
+    host = config[CONF_HOST]
+    port = config[CONF_PORT]
+
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
```suggestion
```

Reply:
Thats used for the global list of players and touching a whole lot of the discovery code, Id rather not go there as its touching @rajlaud code deeper than I need to

Reply:
Can you maybe explain what this is used for better? Because with the move to entry.runtime_data, we use the `hass.data` usually for integration global things. So is this stuff that is only present for 1 entry? Then it should be runtime data, otherwise hass.data

(The reason I'm starting about this now is that you're going to add more stuff, so better get this straight before adding more features :) )

Reply:
Ok yes the stuff I will can go back in enty_data or similar, I can but the line back into media_player and leave that code untouched now as the compromise, but yes you need a view of players over the integrations as they can move server

Reply:
Open comment

Reply:
So I have moved it back into media_player, users and the severs can change the server and library a player is connected via (I have two here at home a test-dev and prod (I maintain an LMS addon for HA to run the LMS server in)

Reply:
Here is how the LMS architecture works. You can have one or more servers on your network (most applications involve only one). You can also have one or more players on your network (most applications have more than one). Traditionally the players were hardware devices but many current applications involve software emulating a player, such as [squeezelite](https://github.com/ralph-irving/squeezelite) running on a raspberry pi.

Once a player connects to a server, the server sends it music and also can control the player (pause, play, etc.). So the way the integration controls the player is to connect to the appropriate server and have it tell the player what to do.

In the simple case (one server) the player automatically connects to that server, but if you have multiple servers you can tell the player which to connect to. The players are uniquely identified by their MAC address (or a user-defined value in the case of a software player) so we want to be able to track the players across servers.

Just catching up on this PR, but happy to help explain the code base if you have questions. A lot has changed in HA since I first took this over, so I'm sure there are more "modern" ways to do some of these things now.

Reply:
And if I had to do a guess we only have players as entities and devices in home assistant and not yet the server?

Reply:
Exactly. It's been that way since I took over the integration to refactor/make async. I never really thought about it, but I see how adding the server might be useful, so you can monitor its status in HA.

Reply:
@rajlaud yes server sensors incoming, I have a very grotty implementation in the LMS addon but to be quite frank its rubbish and I had two choices MQTT or direct HA... and MQTT would be a big for not addon servers.  I have them all working on another fork I need to fold in @joostlek feedback from today and I will stage up another PR.